stock_landed_costs_purchase_auto/models/purchase_order.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    landed_cost_ids = fields.One2many(
+        comodel_name="stock.landed.cost",
+        inverse_name="purchase_id",
+        string="Landed Costs",
+    )
+    landed_cost_number = fields.Integer(compute="_compute_landed_cost_number")
+
+    def _compute_landed_cost_number(self):
+        domain = [("purchase_id", "in", self.ids)]
+        res = self.env["stock.landed.cost"].read_group(
+            domain=domain, fields=["purchase_id"], groupby=["purchase_id"]
+        )
+        landed_cost_dict = {x["purchase_id"][0]: x["purchase_id_count"] for x in res}
+        for item in self:
+            item.landed_cost_number = landed_cost_dict.get(item.id, 0)
+
+    def _prepare_landed_cost_values(self, picking):
+        return {
+            "purchase_id": self.id,
+            "picking_ids": picking.ids if picking else False,
```

Comment:
Remember that this is not valid when preparing a dictionary vals for a m2m, and at this level, `picking` **mustn't** be False. If False, no call to this method should be done:
```suggestion
            "picking_ids": [(4, picking.id)],
```